enhancements/network/multus-services.md

```diff
@@ -0,0 +1,239 @@
+- Headless service (need to more discussion in community for its design)
+
+## Proposal
+
+### Target Service Functionality
+
+We are targeting the following functionality for services in this proposal:
+
+- ClusterIP
+
+#### Service Reachability
+
+The service can be accessed from the pods that can access the target service network. For example, if `ServiceA` is
+created for `net-attach-def1`, then `ServiceA` can be accessed from the pods which has `net-attach-def1` network
+interface and we won't provide any gateway/routers for access from outside of network in this proposal.
+
+#### Cluster IP for Multus
+
+When the service is created with `type: ClusterIP`, then Kubernetes assign cluster virtual IP for the services. User
+can access the services with given virtual IP. The request to virtual IP is automatically replaced with actual
+pods' network interface IP and send to the target services. User needs to make sure reachability to the target
+network, otherwise the request packet will be dropped.
+
+### User Stories
+
+- As an OpenShift developer, I require the ability to segregate my Kubernetes service data plane traffic onto different networks for reasons of security and performance.
+- As an OpenShift developer, I require the ability to segregate my Kubernetes service data plane traffic to a secondary interface that is associated with a different CNI plugin to provide functionality not available in the primary (default network) CNI plugin.
+- As an OpenShift networking administrator, I require the ability to provide access to an internally-facing corporate network and an externally-facing DMZ network, so application developers can choose which network (pod interface) to use for the different types of traffic that make up their application (e.g. web server:external, DB:internal).
+- As an OpenShift network administrator, I require the ability to use one NIC on my hosts for control plane traffic, and additional NICs for data plane traffic.
+
+### Implementation Details/Notes/Constraints
+
+In this enhancement, we will introduce following components (component names may be changed during development/community discussion):
+
+- Multus Service Controller
+- Multus Proxy
+
+The Multus service controller watches all pods' secondary interfaces, services and network attachment definitions and
+it creates `EndpointSlice` for the service. `EndpointSlice` contains secondary network IP address whidch Multus proxy
+watches. Service and EndpointSlice have the label, `service.kubernetes.io/service-proxy-name`, which is defined at
+[kube-proxy APIs](https://pkg.go.dev/k8s.io/kubernetes/pkg/proxy/apis), to make target service out of Kubernetes
+management.
+
+Multus proxy provides forwarding plane for multus-service with iptables at Pod's network namespace. It watches Service
+and EndpointSlice with 'service.kubernetes.io/service-proxy-name: multus-proxy' and creates changes destination to
```

Comment:
this sentence does not read well; I suspect the word `creates` should be deleted.